### PR TITLE
9.2 Achievements board

### LINKED ISSUE #49: 9.2 Create achievements board

### PR DESCRIPTION
**Pull request by Erik142**
Closes #49. Depends on #48, therefore #48 should be merged before this PR.

As previously stated, this builds on #48, and adds a separate Activity for looking at all unlocked achievements. The achievement view can be triggered from the MainActivity by clicking on the icon in the Toolbar. When clicking the icon in the Toolbar, a new activity will be opened, displaying all unlocked achievements:

In the image we have added fake achievement data just to show what the data would actually look like. This sample data is not included in this PR, but if you want to try it out for yourself you can pull the branch to your local repo, then cherry-pick commit 0d973f1 to re-add the sample data.

The user can also click the arrow icon in the Achievements view, or the back button on the device itself, which will lead them back to the map view, MainActivity.

### CODE REVIEW

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.List;
+
+/**
+ * @author Erik Wahlberger
+ * @version 2021-10-13
+ * A fragment representing a list of Items.
+ */
+public class AchievementFragment extends Fragment {
+
+    // TODO: Customize parameter argument names
```
Comment (danbrn) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java:35]: TODO! Important enough to do something about or not important enough to keep? ;)

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    private static final String ARG_COLUMN_COUNT = "column-count";
+    // TODO: Customize parameters
```
Comment (danbrn) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java:37]: TODO!

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+     * Mandatory empty constructor for the fragment manager to instantiate the
+     * fragment (e.g. upon screen orientation changes).
+     */
+    public AchievementFragment() {
+        achievements = new ArrayList<>();
+    }
+
+    // TODO: Customize parameter initialization
```
Comment (danbrn) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/view/achievement/AchievementFragment.java:51]: TODO!

**Review by danbrn — CHANGES_REQUESTED** [review]
Seems to work, and the code looks pretty good. Some of the comments bug me a bit, though... ;)

**Comment by Erik142** [comment]
Haha, I better remove these... Basically the entire AchievementFragment was generated by Android Studio, so I'm basically innocent when it comes to the TODOs... 😆

**Pull request merged**